### PR TITLE
Add BaseXSendFileView

### PR DESCRIPTION
**Pull request by ad-m**

### CODE REVIEW

File: sendfile/views.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from django.views.generic.detail import BaseDetailView
+from sendfile import sendfile
+
+class BaseXSendFileView(BaseDetailView):
```
Comment (moggers87) [inline sendfile/views.py]: I don't think this is a base view, the only things I need to provide can be passed in via kwargs to `as_view`.

Also, a docstring explaining what `file_field` and `send_as_attachment` are would be good.

**Review by moggers87 — CHANGES_REQUESTED** [review]
* There are errors from flake8 - you can run `tox -e flake8` to see them
* There are no tests

**Comment by moggers87** [comment]
Thanks for the PR. There are a couple of issues that I've highlighted in my review, but otherwise this looks good.

**Comment by moggers87** [comment]
@ad-m are you still working on this? There's no way I can merge this without tests.

**Comment by ad-m** [comment]
I will not have time in the coming days. Adding tests to this code requires a significant extension of the tests, as currently no view is tested.

**Comment by moggers87** [comment]
Going close this now, feel free to re-open with tests.

**Pull request closed without merging**